### ISSUE TITLE
[Bug] Firebase 11.7 & Admob 8.7.0 - com.android.build.gradle.tasks.ProcessLibraryManifest$ProcessLibWorkAction

### ISSUE THREAD

**Issue by dcosmin-97** [firebase/firebase-unity-sdk#945]
### [REQUIRED] Please fill in the following fields:

  * Unity editor version: 2022.3.19f1
  * Firebase Unity SDK version: 11.7.0
  * Problematic Firebase Component: Messaging
  * Other Firebase Components in use: Analytics, Crashlytics
  * Additional SDKs you are using: Admob + Mediations
  * Platform you are using the Unity editor on: Windows
  * Platform you are targeting: Android
  * Scripting Runtime: ILCPP

\Library\Bee\Android\Prj\IL2CPP\Gradle\unityLibrary\src\main\AndroidManifest.xml:28:5-87 Error:
	Element property at AndroidManifest.xml:28:5-87 annotated with 'tools:node="removeAll"' cannot have other attributes : xmlns:tools
\Library\Bee\Android\Prj\IL2CPP\Gradle\unityLibrary\src\main\AndroidManifest.xml Error:
	Validation failed, exiting
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files use unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':unityLibrary:processReleaseManifest'.
> A failure occurred while executing com.android.build.gradle.tasks.ProcessLibraryManifest$ProcessLibWorkAction
   > Manifest merger failed with multiple errors, see logs

In my Android Manifest, I have the following:
```
<activity android:name="com.google.firebase.MessagingUnityPlayerActivity" android:configChanges="mcc|mnc|locale|touchscreen|keyboard|keyboardHidden|navigation|orientation|screenLayout|uiMode|screenSize|smallestScreenSize|fontScale|layoutDirection|density" android:screenOrientation="userLandscape" android:launchMode="singleTask" android:resizeableActivity="false" android:hardwareAccelerated="false" android:exported="true">
      <intent-filter>
        <category android:name="android.intent.category.LAUNCHER" />
        <action android:name="android.intent.action.MAIN" />
      </intent-filter>
      <meta-data android:name="unityplayer.UnityActivity" android:value="true" />
      <meta-data android:name="notch_support" android:value="true" />
    </activity>
    <property tools:node="removeAll" xmlns:tools="http://schemas.android.com/tools" />

```

This line is causing issues: property tools:node="removeAll" xmlns:tools="http://schemas.android.com/tools" 

Are there any temporary fixes?
Thank you

**Comment by a-maurice** [comment]
At the top of your AndroidManifest file, do you have something like
`<manifest ... xmlns:tools="http://schemas.android.com/tools">`?

If not, can you add that tools section to the manifest tag, and then remove that part from the property removeAll tag?

**Comment by pogestudio** [comment]
I ran into same problem for firebase 11.7.0, found this thread after much search!

Added to existing project. Most notable packages are ARFoundation and ARCore. 

In the AndroidManifest.xml I have the line
`<manifest ... xmlns:tools="http://schemas.android.com/tools">?`

I did NOT have
`    <property tools:node="removeAll" xmlns:tools="http://schemas.android.com/tools" />`

Just 
`    <property tools:node="removeAll"  />` 

But after I removed this line it compiled and seems to work.

**Comment by Cabskee** [comment]
In addition to removing the <property> block, I had to add a small comment
```xml
	<!-- AnalyticsFixPropertyRemover -->
```
to the bottom of my AndroidManifest.xml so that Firebase didn't try and re-add it each build.

**Comment by batteredhedgehog** [comment]
I also have this problem.  
What are the implications of disabling the AnalyticsFixPropertyRemover?
I cant find information as to why these lines are inserted into AndroidManifest.xml

**Comment by MuhammadWaqasOfficial** [comment]
facing same error. same environment other than unity version 2022.3.20f1

**Comment by JUNE892** [comment]
I also same error (Editor version 2022.3.20f1)

**Comment by a-maurice** [comment]
Hey all,

For some context, the reason for the removeAll line was a problem introduced by one of our Android dependencies.  It added a new tag to its AndroidManifest.xml file, <property ...>.  The problem is that Gradle only added support for the property tag around version 7, and if you are using a version older than that (which Unity 2021 and older do), the build will fail because it doesn't know how to handle the property tag.

What the removeAll is meant to do is remove that property tag from being merged in, and thus prevent older Gradle versions from breaking, but it seems like the generation logic might be missing some edge cases (like the tools namespace not being defined), and 2022.3 + GMA causes some issues that our testing missed.  I will investigate these cases, and see if they can be improved, but as people have pointed out, if the removeAll line is causing problems, it can be removed, and leaving the comment tag `<!-- AnalyticsFixPropertyRemover -->` will prevent it from adding it back in.

**Comment by MuhammadWaqasOfficial** [comment]
> Hey all,
> 
> For some context, the reason for the removeAll line was a problem introduced by one of our Android dependencies. It added a new tag to its AndroidManifest.xml file, <property ...>. The problem is that Gradle only added support for the property tag around version 7, and if you are using a version older than that (which Unity 2021 and older do), the build will fail because it doesn't know how to handle the property tag.
> 
> What the removeAll is meant to do is remove that property tag from being merged in, and thus prevent older Gradle versions from breaking, but it seems like the generation logic might be missing some edge cases (like the tools namespace not being defined), and 2022.3 + GMA causes some issues that our testing missed. I will investigate these cases, and see if they can be improved, but as people have pointed out, if the removeAll line is causing problems, it can be removed, and leaving the comment tag `<!-- AnalyticsFixPropertyRemover -->` will prevent it from adding it back in.

Unity 2022.3 is using gradle version 7, so its not suppose to happen with 2022.3 I guess.
https://docs.unity3d.com/Manual/android-gradle-overview.html

**Comment by JUNE892** [comment]
Everyone, after struggling for over 20 hours over 3 days, I finally succeeded in build it. 
In conclusion, I upgraded the Unity engine version from 2022.3.20f1 to 2023.2.10f1. 
Then the Android build was successful. I hope this helps you.

**Comment by MuhammadWaqasOfficial** [comment]
> Everyone, after struggling for over 20 hours over 3 days, I finally succeeded in build it. In conclusion, I upgraded the Unity engine version from 2022.3.20f1 to 2023.2.10f1. Then the Android build was successful. I hope this helps you.

this will solve 1 problem and introduce many others. as 2022.3 still has many issues. what will happen in 2023.2. :)

**Comment by burakekici** [comment]
Facing with the same error. 
Unity 2022.3.18 + Firebase 11.7.0 + Google MobileAds 8.7.0 + Target API 33.

**Comment by MuhammadWaqasOfficial** [comment]
> Everyone, after struggling for over 20 hours over 3 days, I finally succeeded in build it. In conclusion, I upgraded the Unity engine version from 2022.3.20f1 to 2023.2.10f1. Then the Android build was successful. I hope this helps you.

not working for me. I test on unity 2023.2.10f1. Build failed with same error.

**Comment by a-maurice** [comment]
@MuhammadWaqasOfficial If you are using 2022 or 2023, you can remove the <removeAll> line from the AndroidManifest.xml file, and that should resolve this issue.  Or have you already done so, and it is still causing problems?

I'm going to improve the logic to not add that line if it finds a version of gradle that should support the <property> tag, so if that isn't working for you, it would be good to know.

**Comment by MuhammadWaqasOfficial** [comment]
> @MuhammadWaqasOfficial If you are using 2022 or 2023, you can remove the line from the AndroidManifest.xml file, and that should resolve this issue. Or have you already done so, and it is still causing problems?
> 
> I'm going to improve the logic to not add that line if it finds a version of gradle that should support the tag, so if that isn't working for you, it would be good to know.

yes working with by removing <property tools:node="removeAll" /> line and .
But seeing crashes and ANR's on play console and crashlytics (firebase 11.7.0 (Eternal Dependency Manager 1.2.179) , admob unity 8.7.0 (Eternal Dependency Manager 1.2.177) )( resolving with 1.2.179)
Don't know if its case by firebase or admob.

**Comment by MuhammadWaqasOfficial** [comment]
@dcosmin-97 please confirm if you're using liftoff mediation (5.0.0)? If yes please share all errors. Because in my case its because of liftoff mediation. I dont have to do any suggested solution like removing <property tools:node="removeAll" /> or adding

**Comment by XinqingHappy** [comment]
Unity 2021.3.20f1 + Firebase 11.7.0(FirebaseAnalytics、FirebaseRemoteConfig、FirebaseMessaging)，Same problem。 can anyone tell me the solution?

**Comment by zoranigic** [comment]
Solution is to replace

`    
    <property tools:node="removeAll" />`

 with 

`    
`

**Comment by DmitryKoshmanFP** [comment]
Guys, i think that im found a workaround. Just move `xmlns:tools="http://schemas.android.com/tools"` attribute to manifest element. This solved my issue.

```
<manifest ... xmlns:tools="http://schemas.android.com/tools>
   <application ...>
   
      <!--This was added by the AnalyticsFixPropertyRemover. If you want to prevent the generation of this, have "AnalyticsFixPropertyRemover" included in a comment-->
      <property tools:node="removeAll" />
   
   </application>
</manifest>
```

**Comment by tienassassin** [comment]
Unity 2022.3.16f1, Firebase 11.7.0 (Analytics, Crashlytics, RemoteConfig), same problem. Hope there is a solution soon 😭

**Comment by 37wjy** [comment]
no solution work on unity 2022.3.20f1 

Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/AndroidManifest.xml:42:5-87 Error:
	Element property at AndroidManifest.xml:42:5-87 annotated with 'tools:node="removeAll"' cannot have other attributes : xmlns:tools
Library/Bee/Android/Prj/IL2CPP/Gradle/unityLibrary/src/main/AndroidManifest.xml Error:
	Validation failed, exiting

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':unityLibrary:processReleaseManifest'.
> A failure occurred while executing com.android.build.gradle.tasks.ProcessLibraryManifest$ProcessLibWorkAction
   > Manifest merger failed with multiple errors, see logs

**Comment by DmitryKoshmanFP** [comment]
@37wjy in your case, you are using unity, which already have gradle which supports property tags. Just follow @a-maurice or @zoranigic solutions.

**Comment by a-maurice** [comment]
We just released [11.8.0](https://github.com/firebase/firebase-unity-sdk/releases/tag/v11.8.0), which improves the logic a bit by checking which version of gradle is being used, and if it is a new enough version, then it won't add the removeAll logic.  That should resolve this issue for Unity 2022, without requiring you to adjust your AndroidManifest.xml manually anymore.

I'm going to close this issue, since this should resolve the problem that a lot of people were hitting with this.
